dbos/_sys_db.py

```diff
@@ -1351,12 +1358,6 @@ def start_queued_workflows(self, queue: "Queue", executor_id: str) -> List[str]:
 
             max_tasks = float("inf")
             if queue.worker_concurrency is not None:
-                # Worker local concurrency limit should always be >= running_tasks_for_this_worker
-                # This should never happen but a check + warning doesn't hurt
-                if running_tasks_for_this_worker > queue.worker_concurrency:
-                    dbos_logger.warning(
-                        f"Number of tasks on this worker ({running_tasks_for_this_worker}) exceeds the worker concurrency limit ({queue.worker_concurrency})"
-                    )
```

Comment:
This check & warning are now inacurate because DBOS Cloud will "reassign" the workflows (by writing to `dbos.workflow_status.executor_id`) without resetting `started_at_epoch_ms`.

This means that the read above will "think" the worker is already running enough tasks -- during recovery. This is only a transient situation because `clear_queue_assignment` now resets `started_at_epoch_ms`, and the worker will lose maybe one or two dequeue iterations.